kratos/python_scripts/compare_two_files_check_process.py

```diff
@@ -139,52 +152,54 @@ def __CompareResultsBlock(self, lines1, lines2, current_index):
 
         return current_index+2 # directly incrementing to get the new result label
 
+    def __CompareDatFile(self):
+        lines_ref, lines_out = self.__GetFileLines()
 
-def ConvertStringToListFloat(line, space = " ", endline = ""):
-    list_values = []
-    string_values = (line.replace(endline,"")).split(space)
-    for string in string_values:
-        list_values.append(float(string))
+        # assert headers are the same
+        while lines_ref[0].lstrip()[0] == '#' or lines_out[0].lstrip()[0] == '#':
+            self.assertTrue(lines_ref.pop(0) == lines_out.pop(0))
 
-    return list_values
+        # assert values are equal up to given tolerance
+        for line_ref, line_out in zip(lines_ref, lines_out):
+            for v1, v2 in zip(line_ref.split(), line_out.split()):
+                self.assertAlmostEqual(float(v1),
+                                       float(v2),
+                                       self.decimal_places)
 
-def _ReadVertices(input_file1, input_file2, dimension):
-    with open(input_file1,'r') as f1, open(input_file2,'r') as f2:
-        lines1 = f1.readlines()
-        lines2 = f2.readlines()
+    def __CompareMeshVertivesFile(self):
+        lines_ref, lines_out = self.__GetFileLines()
 
         numline = 0
-        for line1 in lines1:
+        for line1 in lines_ref:
             numline += 1
 
             if("Vertices" in line1):
-                line = lines1[numline]
+                line = lines_ref[numline]
                 nvertices = int(line)
                 numline += 1
                 break
 
         error = 0.0
         for i in range(numline, nvertices + numline):
-            tmp1 = ConvertStringToListFloat(lines1[i], "", "\n")
-            tmp2 = ConvertStringToListFloat(lines2[i], "", "\n")
+            tmp1 = ConvertStringToListFloat(lines_ref[i], "", "\n")
+            tmp2 = ConvertStringToListFloat(lines_out[i], "", "\n")
             if (dimension == 2):
                 error += ((tmp1[0] - tmp2[0])**2.0 + (tmp1[1] - tmp2[1])**2.0)**(0.5)
```

Comment:
guys i have a serious problem to understand why you use a double exponent 2.0 

I am very convinced that you should use 2 (not 2.0 as exponent). Also DO NOT use exponent 0.5 for sqrt, use the sqrt instead which goes through a specialized pathway!!

Reply:
makes sense to me, this was written by @loumalouomega 

@loumalouomega I will change it if you are ok with it